doc/sphinx/proof-engine/tactics.rst

```diff
@@ -460,16 +460,22 @@ Examples:
 Occurrence sets and occurrence clauses
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-An occurrence clause is a modifier to some tactics that obeys the
+An occurrence clause (:n:`in @in_clause`) is a modifier to some tactics that obeys the
 following syntax:
 
-  .. productionlist:: coq
-     occurrence_clause : in `goal_occurrences`
-     goal_occurrences : [`ident` [`at_occurrences`], ... , `ident` [`at_occurrences`] [|- [* [`at_occurrences`]]]]
-                      : * |- [* [`at_occurrences`]]
-                      : *
-     at_occurrences : at `occurrences`
-     occurrences     : [-] `num` ... `num`
+  .. insertprodn in_clause hypident
+
+  .. prodn::
+     in_clause ::= @in_clause
+     | * {? at @occs_nums }
+     | * |- {? @concl_occ }
+     | {*, @hypident_occ } |- {? @concl_occ }
+     | {*, @hypident_occ }
```

Comment:
We can make this grammar simpler, closer to what the `productionlist` was showing. For instance

```
in_clause ::= {+, @hypident {? at @occurrences } } {? |- {? * {? at @occurrences } } }
| * |- {? * {? at @occurrences } }
| * {? at @occurrences }
```